utility_analysis_new/utility_analysis.py

```diff
@@ -102,27 +102,37 @@ def perform_utility_analysis(col,
     # accumulators : (aggregate_metrics)
 
     def pack_metrics(aggregate_metrics) -> List[metrics.AggregateMetrics]:
-        if public_partitions is None:
-            # aggregate_metrics has format  [PartitionSelectionMetrics,
-            # AggregateErrorMetrics, PartitionSelectionMetrics ...]
-            # Each consecutive pair PartitionSelectionMetrics and
-            # AggregateErrorMetrics correspond to one Utility Analysis
-            # configuration.
-            return [
-                metrics.AggregateMetrics(aggregate_metrics[i + 1],
-                                         aggregate_metrics[i])
-                for i in range(0, len(aggregate_metrics), 2)
-            ]
-        return [
-            metrics.AggregateMetrics(aggregate_error_metrics)
-            for aggregate_error_metrics in aggregate_metrics
-        ]
+        # aggregate_metrics is a flat list of PartitionSelectionMetrics and
+        # AggregateErrorMetrics with options.n_parameters sequential
+        # configurations of metrics. Each AggregateErrorMetrics within a
+        # configuration correspond to a different aggregation.
+        metrics_per_config = len(aggregate_metrics) // options.n_parameters
+        return_list = []
+        for i in range(0, options.n_parameters):
+            packed_metrics = metrics.AggregateMetrics()
+            for j in range(i * metrics_per_config,
+                           (i + 1) * metrics_per_config):
+                _populate_packed_metrics(packed_metrics, aggregate_metrics[j])
+            return_list.append(packed_metrics)
+        return return_list
 
     return backend.map(aggregates, pack_metrics,
                        "Pack metrics from the same run")
     # (aggregate_metrics)
 
 
+# Sets the appropriate field of packed_metrics with aggregate_error_metric
```

Comment:
Nit: Python docstring style
```
def funcname() ...
"""1 line description""
```
or
```
def funcname() ...
"""1 line description

<optional long description>

Args:
  arg1: ..

Returns:
""
```

Here for me it looks like 1 line description is the best, something like
"Sets the the appropriate packed_metrics field with 'metric' according to 'metric' type. " (and to rename aggregate_error_metric -> metric)